### ISSUE TITLE
Resolve texts to ArXiv IDs where possible, and augment our metadata with it

### ISSUE THREAD

**Issue by JonathanReeve** [JonathanReeve/data-ethics-literature-review#28]
This is a sub-task of #21, and a sibling task of #27. 

Let's use the [ArXiv API](https://arxiv.org/help/api/index) to resolve texts, and use the data retrieved from that API to augment our own bibliographic data. 

The easy ones will be ones that have ArXiv IDs: 

```
<https://data-ethics.tech/text/#item_28> a z:UserItem ;
    res:resource [ a bibo:Note ;
            z:note "<p>ArXiv Preprint ArXiv:1609.07236.</p>" ],
        [ a bibo:Note ;
            z:note "<p>very short, read first</p>" ],
        [ a bibo:Note ;
            z:note "<p>What Is Technology:</p>" ],
        [ a bibo:Note ;
            z:note "<p>available online at</p>" ],
        [ a bibo:Note ;
            z:note "<p>from</p>" ] .
```

(That example is very messy.) 

This will make #14 a lot easier.

**Comment by JonathanReeve** [comment]
@sy2657, can you take this one?

**Comment by JonathanReeve** [comment]
[Here is the beginning of an example where I'm querying the CrossRef API for bibliographic data that I'm then using to augment the graph.](https://github.com/JonathanReeve/data-ethics-literature-review/blob/1ed14be780f74c2c04a006e037145320516893df/turtleize/enhanceBibliography.py).

**Comment by sy2657** [comment]
Ok, sure.

**Comment by JonathanReeve** [comment]
@sy2657, just leave a note here if you have any questions about this issue, or how best to approach it.

**Comment by JonathanReeve** [comment]
Here's a breakdown of how I imagine this would go:

 - For each reading in the coursesAndTexts.ttl graph,
 - See if it has an ArXiv ID first,
   + if it does, just get all the ArXiv's metadata for that paper,
 - If it doesn't have an ArXiv ID, query the API for its title and author, or whatever other fields return the most accurate results. 
 - Maybe do some kind of check using Levenshtein distance to make sure the search result has the same title as the queried title
 - Merge the search result metadata with our metadata for that article

@Zhuohan-Amber, the process is pretty much the same for #27, only without the semantic scholar IDs. There, the DOI is the thing that gets us the furthest.

**Comment by ssy248** [comment]
Hi, 
 I ran your code for queryCrossRef on a small subset of coursesAndTexts.ttl and it did not print out any data...

For 


deUniversity:victoria-university-of-wellington a ccso:University ;
    owl:sameAs wikidata:Q1150437 ;
    ccso:legalName "Victoria University of Wellington" .

<https://data-ethics.tech/text/#item_28> a z:UserItem ;
    res:resource [ a bibo:Note ;
            z:note "<p>ArXiv Preprint ArXiv:1609.07236.</p>" ],
        [ a bibo:Note ;
            z:note "<p>very short, read first</p>" ],
        [ a bibo:Note ;
            z:note "<p>What Is Technology:</p>" ],
        [ a bibo:Note ;
            z:note "<p>available online at</p>" ],
        [ a bibo:Note ;
            z:note "<p>from</p>" ] .

<https://data-ethics.tech/text/#item_29> a z:UserItem ;
    res:resource [ a bibo:Note ;
            z:note "<p>by Madeleine Claire Elish and Tim Hwang (Quartz, 2015</p>" ],
        [ a bibo:Note ;
            z:note "<p>Available at:</p>" ],
        [ a bibo:Note ;
            z:note "<p>group (TBD</p>" ],
        [ a bibo:Note ;
            z:note "<p>Available at</p>" ],
        [ a bibo:Note ;
            z:note "<p>content warning: adult, possibly not-safe-for-work language</p>" ] .

<https://data-ethics.tech/text/#item_3> a z:UserItem ;
    res:resource [ a bibo:Note ;
            z:note "<p>Kindle epublication.</p>" ],
        [ a bibo:Note ;
            z:note "<p>Up to “Open Peer Commentary”</p>" ],
        [ a bibo:Note ;
            z:note "<p>Imagining the Internet Robin Mansell</p>" ],
        [ a bibo:Note ;
            z:note "<p>Library:</p>" ],
        [ a bibo:Note ;
            z:note "<p>Available at:</p>" ] .

Also how do I save the metadata ? 
I save (metadata) Title, Authors, Abstract,  Comments, Report-no, Category,  Journal-ref, DOI, MSC-class, ACM-class in some format ?

**Comment by JonathanReeve** [comment]
Yep, that code isn't working yet. It's just the beginnings of an example. The idea is just: go through each text (`z:UserItem`), find more metadata about the text, and then add it to the graph. 

The text metadata format is [bibliontology](http://www.bibliontology.com/specification.html#). [Here's an example for an article](http://www.bibliontology.com/content/article.html), and [here's one for a book](http://www.bibliontology.com/content/book.html). Bibliontology uses Dublin Core (`dcterms`) for things like titles and authors. 

So if you have variables `title` and `author`, and others, to add to the graph, you can add it with something like: 

```
g.add((item, dcterms:title, title))
g.add((item, dcterms:creator, author))
```

where `item` is the `z:UserItem`.
